File: VB-Examples/07_Conversion/LoadSaveEtAndETT.vb
Imports Spire.Xls

Namespace LoadSaveEtAndETT
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of Workbook
            Dim workbook As New Workbook()

            ' Load the ET file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\Sample-et.et")
            ' Uncomment the line below to load an ETT file instead
            ' workbook.LoadFromFile("..\..\..\..\..\..\Data\Sample-ett.ett");

            ' Save the workbook to an ET file with the specified output file name
            workbook.SaveToFile("result.et", FileFormat.ET)
            ' Uncomment the line below to save as ETT instead
            ' workbook.SaveToFile("result.ett", FileFormat.ETT);
            ' Release the resources used by the workbook
            workbook.Dispose()

            'view the document
            ExcelDocViewer("result.et")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub


	End Class
End Namespace

File: CS-Examples/07_Conversion/LoadSaveEtAndETT.cs
using System;
using System.Windows.Forms;

using Spire.Xls;

namespace LoadSaveEtAndETT
{
	public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
		{
            //create a workbook
            Workbook workbook = new Workbook();

            //load .et or .ett file 
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\Sample-et.et");
            //workbook.LoadFromFile(@"..\..\..\..\..\..\Data\Sample-ett.ett");

            //save to .et or .ett file
            workbook.SaveToFile("result.et", FileFormat.ET);
            //workbook.SaveToFile("result.ett", FileFormat.ETT);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //view the document
            ExcelDocViewer("result.et");
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }


	}
}
